docs/design/design_trivy_file_system_scanner.md

```diff
@@ -0,0 +1,197 @@
+on workloads or on service accounts used by the workloads, and passes them down to Trivy executable as `TRIVY_USERNAME`
+and `TRIVY_PASSWORD` environment variables.
+
+Since ImagePullSecrets are not the only way to provide registry credential, the following alternatives are not
+currently supported by Starboard:
+1. Pre-pulled images
+2. [Configuring nodes to authenticate to a private registry]
+3. Vendor-specific or local extension. For example, methods described on [AWS ECR Private registry authentication].
+
+Even though we could resolve some of above-mentioned limitations with hostPath volume mounts to the container runtime
+socket, it would have its own disadvantages that we are trying to avoid. For example, more permissions to schedule scan
+Jobs and additional information about cluster's infrastructure such as location of the container runtime socket. 
+
+## Solution
+
+### TL;DR;
+
+Use Trivy filesystem scanning to scan container images. The main idea, which is discussed in this proposal, is to
+schedule a scan Job on the same cluster node where the scanned workload. This allows Trivy to scan a filesystem of
+the container image which is already cached on that node without pulling the image from a remote registry. What's more,
+Trivy will scan container images from private registries without providing registry credentials (as ImagePullSecret or
+in any other proprietary way).
+
+### Deep Dive
+
+To scan a container image of a given K8s workload Starboard will create a corresponding container of a scan Job and
+override its entrypoint to invoke Trivy filesystem scanner.
+
+This approach requires Trivy executable to be downloaded and made available to the entrypoint. We'll do that by adding
+the init container to the scan Job. Such init container will use the Trivy container image to copy Trivy executable out
+to the emptyDir volume, which will be shared with the other containers.
+
+Another init container is required to download Trivy vulnerability database and save it to the mounted shared volume.
+
+Finally, the scan container will use shared volume with the Trivy executable and Trivy database to perform the actual
+filesystem scan. (See the provided [Example](#example) to have a better idea of all containers defined by a scan Job and
+how they share data via the emptyDir volume.)
+
+> Note that the second init container is required in [Standalone] mode, which is the only mode supported by Trivy
+> filesystem scanner at the time of writing this proposal.
+
+We further restrict scan Jobs to run on the same node where scanned Pod is running and never pull images from remote
+registries by setting the `ImagePullPolicy` to `Never`. To determine the node for a scan Job Starboard will list active
+Pods controlled by the scanned workload. If the list is not empty it will take the node name from the first Pod,
+otherwise it will ignore the workload.
```

Comment:
This may run afoul of this admission controller https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages

Reply:
That's a very good point. We've merge this PR already but I opened #900 to add you comments as remark.